bdk_chain/src/tx_graph.rs

```diff
@@ -167,38 +170,87 @@ impl TxGraph {
         })
     }
 
-    /// Iterate over all full transactions in the graph
+    /// Iterate over all full transactions in the graph.
     pub fn full_transactions(&self) -> impl Iterator<Item = &Transaction> {
         self.txs.iter().filter_map(|(_, tx)| match tx {
             TxNode::Whole(tx) => Some(tx),
             TxNode::Partial(_) => None,
         })
     }
 
+    /// Iterate over all partial transactions (outputs only) in the graph.
     pub fn partial_transactions(&self) -> impl Iterator<Item = (Txid, &BTreeMap<u32, TxOut>)> {
         self.txs.iter().filter_map(|(txid, tx)| match tx {
             TxNode::Whole(_) => None,
             TxNode::Partial(partial) => Some((*txid, partial)),
         })
     }
 
-    /// Return an iterator of conflicting txids, where the first field of the tuple is the vin of
-    /// the original tx in which the txid conflicts.
-    pub fn conflicting_txids<'g>(
+    /// Creates an iterator that both filters and maps descendants of the given `txid`.
+    ///
+    /// When the supplied closure returns `None`, spend paths from the given descendant are not
+    /// explored, and the descendant will not be yielded by the iterator.
+    pub fn filter_map_descendants<'g, T>(
+        &'g self,
+        txid: Txid,
+        filter_map: impl FnMut(OutPointSpends, Txid) -> Option<T> + 'g,
```

Comment:
Why does the closure need to be a filter map? Can't it just do filtering.

I think what would make sense to me is:

```
fn walk_descendants(&self, txid: Txid, should_walk: FnMut(&Transaction, u32) -> bool) -> impl Iterator<Item=&Transaction> {
...
}
```

Reply:
I think it is more helpful and flexible as a filter map. For example, when `ChainGraph` detects conflicts of a given transaction, we only want to include conflicts that are in `SparseChain` (filter). Additionally, we want the conflict to be returned with the `ChainPosition` (map).

Reply:
Resolved in DMs, @LLFourn agrees with me.